cli/pkg/backend/aws/k8s_compute_backend.go

```diff
@@ -208,7 +215,68 @@ func (k8s *K8SComputeBackend) PrintLogs(ctx context.Context, stackName string, s
 	return nil
 }
 
-func (b *K8SComputeBackend) RunTask(ctx context.Context, taskDefArn string, launchType config.LaunchType) error {
+func (k8s *K8SComputeBackend) RunTask(ctx context.Context, taskDefArn string, launchType config.LaunchType) error {
 	// TODO: not implemented
 	return errors.New("not implemented")
 }
+
+func (k8s *K8SComputeBackend) Shell(ctx context.Context, stackName string, serviceName string) error {
+	deploymentName := fmt.Sprintf("%s-%s", stackName, serviceName)
+	labelSelector := v1.LabelSelector{MatchLabels: map[string]string{"app": deploymentName}}
+	pods, err := k8s.ClientSet.CoreV1().Pods(k8s.HappyConfig.K8SConfig().Namespace).List(ctx, v1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve a list of pods for deployment %s", deploymentName)
+	}
+	if len(pods.Items) == 0 {
+		return errors.New("No matching pods found")
+	}
+
+	podName := pods.Items[0].Name
+
+	pod, err := k8s.ClientSet.CoreV1().Pods(k8s.HappyConfig.K8SConfig().Namespace).Get(ctx, pods.Items[0].Name, v1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve pod information for %s", podName)
+	}
+
+	if len(pod.Spec.Containers) > 1 {
+		return errors.Errorf("There's more than one container in a pod '%s'", podName)
+	}
+
+	containerName := pod.Spec.Containers[0].Name
+
+	logrus.Infof("Found %d matching pods. Opening a TTY tunnel into pod '%s', container '%s'", len(pods.Items), podName, containerName)
+
+	req := k8s.ClientSet.CoreV1().RESTClient().Post().Resource("pods").Name(pod.Name).Namespace(pod.Namespace).SubResource("exec").Param("container", containerName)
+
+	eo := &corev1.PodExecOptions{
+		Container: containerName,
+		Command:   strings.Fields("sh -c /bin/sh"),
+		Stdout:    true,
+		Stdin:     true,
+		Stderr:    false,
+		TTY:       true,
+	}
+	req.VersionedParams(eo, scheme.ParameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(k8s.rawConfig, http.MethodPost, req.URL())
+	if err != nil {
+		return errors.Wrapf(err, "Unable to create a SPDY executor")
+	}
+
+	stdin, stdout, stderr := dockerterm.StdStreams()
+	streamOptions := remotecommand.StreamOptions{
+		Stdout: stdout,
+		Stderr: stderr,
+		Tty:    true,
+	}
+	streamOptions.Stdin = stdin
```

Comment:
Can stdin be set in the statement above?
```suggestion
	streamOptions := remotecommand.StreamOptions{
		Stdout: stdout,
		Stderr: stderr,
		Stdin:  stdin
		Tty:    true,
	}
```